proposals/threads/Overview.md

```diff
@@ -48,8 +48,48 @@ agent's cluster will have access to the additional linear memory.
 When a module has an imported linear memory, its data segments are copied into
 the linear memory when the module is instantiated.
 
-When the imported linear memory is shared, the writes are non-atomic and
-are not ordered.
+When linear memory is not shared, the ordering of data segment initialization is
+not observable. When linear memory is shared, it is possible for another module
```

Comment:
Yeah I think it's simpler to just specify it for both. Even if non-observable, the mental model is simpler.